packages/sku/src/context/defaultCompilePackages.ts

```diff
@@ -16,10 +17,27 @@ let detectedCompilePackages: string[] = [];
 // package manager. In either case, we can't correctly detect compile packages.
 if (rootDir) {
   try {
+    const pnpmVirtualStorePath = path.join(
+      toPosixPath(rootDir),
+      'node_modules/.pnpm',
+    );
+    const hasPnpmVirtualStore = existsSync(pnpmVirtualStorePath);
+
+    if (hasPnpmVirtualStore && !isPnpm) {
+      banner(
+        'error',
+        'pnpm virtual store found, but another package manager is in use',
+        [
+          'Please use pnpm to build your project or remove the `node_modules/.pnpm` directory.',
+          'Running commands with a different package manager may cause unexpected behaviour.',
+        ],
+      );
+    }
```

Comment:
Suggestions on better wording for this are welcome

Reply:
Might be too long now, but a brief explanation could make the error seem less mysterious and more obvious.

```suggestion
      banner(
        'error',
        'pnpm virtual store found, but another package manager is in use',
        [
          'Please use pnpm to build your project or remove the `node_modules/.pnpm` directory.',
          'Different package managers expect different `node_modules` structures.',
          'Running commands with a different package manager may cause unexpected behaviour.',
        ],
      );
    }
```